Add data retention section to dbt MCP server docs (#9358)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-ai/about-mcp.md

- [Cursor](/docs/dbt-ai/integrate-mcp-cursor)
- [VS Code](/docs/dbt-ai/integrate-mcp-vscode).

## Data retention

> The dbt MCP server doesn't store or retain any production data or job run results itself. It's a read-only access layer that reads metadata, job results, and <Constant name="semantic_layer"/> data from the <Constant name="dbt_platform"/> in real time when a tool is called.

Reviewer: ```suggestion
The dbt MCP server doesn't store or retain any production data or job run results. It's a read-only access layer that reads metadata, job results, and <Constant name="semantic_layer"/> data from the <Constant name="dbt_platform"/> in real time when a tool is called.
```

> How long job runs and artifacts are available to query depends on your [dbt platform data retention policy](https://www.getdbt.com/security), not the MCP server. As long as a job or artifact exists in dbt platform, the MCP server can access it.

Reviewer: ```suggestion
Your [dbt platform data retention policy](https://www.getdbt.com/security) determines how long job runs and artifacts are available, not the MCP server. As long as a job or artifact exists in dbt platform, the MCP server can read it.

```
